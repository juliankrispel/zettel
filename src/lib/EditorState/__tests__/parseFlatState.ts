import parseFlat from '../rawToFlat'

describe('parseFlatState', () => {
  test('parses raw doc', () => {
    const raw = {
      text: '[Hello there world]',
      ranges: [],
      meta: {}
    }

    expect(parseFlat(raw)).toMatchSnapshot()
  })

  test('parses invalid doc as empty', () => {
    const raw = {
      text: 'Hello there world',
      ranges: [],
      meta: {}
    }

    expect(parseFlat(raw)).toEqual({ value: [], meta: {} })
  })

  test('skips invalid bits', () => {
    const raw = {
      text: 'Hello [there] world',
      ranges: [],
      meta: {}
    }

    expect(parseFlat(raw)).toMatchSnapshot()
  })
  
  test('skips invalid nested text', () => {
    const raw = {
      text: '[1[2[3]hello]]',
      ranges: [],
      meta: {}
    }

    expect(parseFlat(raw)).toEqual({
      value: [ { char: '[', metaKeys: [] },
        { char: '1', metaKeys: [] },
        { char: '[', metaKeys: [] },
        { char: '2', metaKeys: [] },
        { char: '[', metaKeys: [] },
        { char: '3', metaKeys: [] },
        { char: ']', metaKeys: [] },
        { char: ']', metaKeys: [] },
        { char: ']', metaKeys: [] } ],
     meta: {} })
  })
})