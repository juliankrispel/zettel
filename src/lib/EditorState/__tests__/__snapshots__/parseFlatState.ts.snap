// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseFlatState parses raw doc 1`] = `
Object {
  "meta": Object {},
  "value": Array [
    Object {
      "char": "[",
      "metaKeys": Array [],
    },
    Object {
      "char": "H",
      "metaKeys": Array [],
    },
    Object {
      "char": "e",
      "metaKeys": Array [],
    },
    Object {
      "char": "l",
      "metaKeys": Array [],
    },
    Object {
      "char": "l",
      "metaKeys": Array [],
    },
    Object {
      "char": "o",
      "metaKeys": Array [],
    },
    Object {
      "char": " ",
      "metaKeys": Array [],
    },
    Object {
      "char": "t",
      "metaKeys": Array [],
    },
    Object {
      "char": "h",
      "metaKeys": Array [],
    },
    Object {
      "char": "e",
      "metaKeys": Array [],
    },
    Object {
      "char": "r",
      "metaKeys": Array [],
    },
    Object {
      "char": "e",
      "metaKeys": Array [],
    },
    Object {
      "char": " ",
      "metaKeys": Array [],
    },
    Object {
      "char": "w",
      "metaKeys": Array [],
    },
    Object {
      "char": "o",
      "metaKeys": Array [],
    },
    Object {
      "char": "r",
      "metaKeys": Array [],
    },
    Object {
      "char": "l",
      "metaKeys": Array [],
    },
    Object {
      "char": "d",
      "metaKeys": Array [],
    },
    Object {
      "char": "]",
      "metaKeys": Array [],
    },
  ],
}
`;

exports[`parseFlatState skips invalid bits 1`] = `
Object {
  "meta": Object {},
  "value": Array [
    Object {
      "char": "[",
      "metaKeys": Array [],
    },
    Object {
      "char": "t",
      "metaKeys": Array [],
    },
    Object {
      "char": "h",
      "metaKeys": Array [],
    },
    Object {
      "char": "e",
      "metaKeys": Array [],
    },
    Object {
      "char": "r",
      "metaKeys": Array [],
    },
    Object {
      "char": "e",
      "metaKeys": Array [],
    },
    Object {
      "char": "]",
      "metaKeys": Array [],
    },
  ],
}
`;
